Eevee.Sleep.Bot/Handlers/InteractionHandler.cs

```diff
@@ -116,4 +119,25 @@ await interaction
                 throw new ArgumentException($"Unhandled modal ID: {modalId}");
         }
     }
+
+    private static async Task OnButtonClicked(SocketMessageComponent component)
+    {
+        var customId = component.Data.CustomId;
+        if (!customId.StartsWith("role"))
+        {
+            return;
+        }
+
+        await component.RespondAsync("You have been given the role!", ephemeral: true);
+
+        var roleId = ulong.Parse(customId.Replace("role", ""));
+        var user = (SocketGuildUser)component.User;
```

Comment:
Don't assume `component.User` is `SocketGuildUser`, use something like `interaction.Type is InteractionType.ApplicationCommand` (example from the repo) to check for the type.

if it doesn't match, see how it should be handled (throw error? ignore?).

Reply:
tried to throw error

[54d6324](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/54d63244d3ed4a4ae211059346051d57d543452c)

Reply:
Commented in the commit.